src/Builder/PackagesBuilder.php

```diff
@@ -77,6 +79,38 @@ public function dump(array $packages): void
             $repo['includes'] = $this->dumpPackageIncludeJson($packagesByName, $this->includeFileName);
         }
 
+        // Composer 2.0 format
+        $metadataUrl = 'p2/%package%.json';
+        if (!empty($this->config['homepage'])) {
+            $repo['metadata-url'] = parse_url(rtrim($this->config['homepage'], '/'), PHP_URL_PATH) . '/' . $metadataUrl;
+        } else {
+            $repo['metadata-url'] = $metadataUrl;
+        }
+
+        foreach ($packagesByName as $packageName => $versionPackages) {
+            $stableVersions = [];
+            $devVersions = [];
+            foreach ($versionPackages as $version => $versionConfig) {
+                if ('stable' === VersionParser::parseStability($versionConfig['version'])) {
```

Comment:
This is not correct, dev versions are only those with 'dev' stability, alpha/beta/etc goes in the other files with all the tags (non-dev versions, not really "stable" versions).

Reply:
The idea of the split is that `dev` versions are invalidated at each push to the repo (well, at least one of them). Tags are immutable, and the non-dev file will then be updated only each time you do a release, which is much less often than each push.

Reply:
Thanks for the review and the explanation. The code has been updated.